indexing-hadoop/src/main/java/org/apache/druid/indexer/InputRowSerde.java

```diff
@@ -190,7 +215,7 @@ public void serialize(ByteArrayDataOutput out, Object value)
     @Override
     public Long deserialize(ByteArrayDataInput in)
     {
-      return in.readLong();
+      return isNullByteSet(in) ? null : in.readLong();
```

Comment:
Perhaps it would be better to use a functional programming style here.

```java
return Optional.ofNullable(in)
               .filter(InputRowSerde::isNotNullByteSet)
               .map(ByteArrayDataInput::readLong)
               .get();
```

Reply:
Eh, I sort of prefer it the way it currently is, seems clearer to me, is there any reason it would be better other than preference?

Reply:
Alright, I prefer the functional style because it makes the code more readable. If we don't use `Optional`, then we need add the `@Nullable` annotation for this method. It's up to you. :sweat_smile:

Reply:
I think I prefer the above style. Maybe I'm misunderstanding me, but I also think the `get()` would cause the code to throw if the null byte is set.

I'll add `@Nullable` annotations to these deserialize methods

Reply:
> but wouldn't the get() cause the code to throw if the null byte is set?

@ferristseng If the null byte is set, then `get` will return a `null` value. What you describe should be the `orElseThrow` function. Thanks for your contribution.